rasgoql/rasgoql/data/postgres.py

```diff
@@ -0,0 +1,630 @@
+            override schema
+        """
+        select_clause = (
+            "SELECT TABLE_NAME, "
+            "TABLE_CATALOG||'.'||TABLE_SCHEMA||'.'||TABLE_NAME AS FQTN, "
+            "CASE TABLE_TYPE WHEN 'BASE TABLE' THEN 'TABLE' ELSE TABLE_TYPE END AS TABLE_TYPE"
+        )
+        from_clause = " FROM INFORMATION_SCHEMA.TABLES "
+        if database:
+            from_clause = f" FROM {database.upper()}.INFORMATION_SCHEMA.TABLES "
+        where_clause = f" WHERE TABLE_SCHEMA = '{schema.upper()}'" if schema else ""
+        sql = select_clause + from_clause + where_clause
+        return self.execute_query(sql, response='df', acknowledge_risk=True)
+
+    def preview(
+            self,
+            sql: str,
+            limit: int = 10
+        ) -> pd.DataFrame:
+        """
+        Returns 10 records into a pandas DataFrame
+
+        Params:
+        `sql`: str:
+            SQL statment to run
+        `limit`: int:
+            Records to return
+        """
+        return self.execute_query(
+            f'{sql} LIMIT {limit}',
+            response='df',
+            acknowledge_risk=True
+        )
+
+    def save_df(
+            self,
+            df: pd.DataFrame,
+            fqtn: str,
+            method: str = None
+        ) -> str:
+        """
+        Creates a table in Postgres from a pandas Dataframe
+
+        Params:
+        `df`: pandas DataFrame:
+            DataFrame to upload
+        `fqtn`: str:
+            Fully-qualied table name (database.schema.table)
+            Name for the new table
+        `method`: str
+            Values: [append, replace]
+            when this table already exists in your DataWarehouse,
+            pass append: to add dataframe rows to it
+            pass replace: to overwrite it with dataframe rows
+                WARNING: This will completely overwrite data in the existing table
+        """
+        if method:
+            method = check_write_method(method)
+        fqtn = magic_fqtn_handler(fqtn, self.default_namespace)
+        db, schema, table = parse_fqtn(fqtn)
+        table_exists = self._table_exists(fqtn)
+        if table_exists and not method:
+            msg = f"A table named {fqtn} already exists. " \
+                   "If you are sure you want to write over it, pass in " \
+                   "method='append' or method='replace' and run this function again"
+            raise TableConflictException(msg)
+        try:
+            cleanse_sql_dataframe(df)
+            # If the table does not exist or we've received instruction to replace
+            # Issue a create or replace statement before we insert data
+            if not table_exists or method == 'REPLACE':
+                create_stmt = generate_dataframe_ddl(df, fqtn)
+                self.execute_query(create_stmt, response='None', acknowledge_risk=True)
+            # TODO: this is nasty, but doing this in SQLAlchemy is pretty nasty tbh
+            df.to_sql(
```

Comment:
lol so scrappy, i love the killer instinct. I think we have some abstractions in pyrasgo we may be able to steal to help here.